File: app/src/main/java/com/damhoe/scoresheetskat/app/about/AboutFragment.java
package com.damhoe.scoresheetskat.app.about;

import android.os.Bundle;

import androidx.annotation.ColorInt;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.damhoe.scoresheetskat.DatabaseInfo;
import com.damhoe.scoresheetskat.MainActivity;
import com.damhoe.scoresheetskat.R;
import com.damhoe.scoresheetskat.databinding.FragmentAboutBinding;
import com.damhoe.scoresheetskat.shared_ui.utils.InsetsManager;
import com.google.android.material.color.MaterialColors;


public class AboutFragment extends Fragment {

    FragmentAboutBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = DataBindingUtil.inflate(inflater,
                R.layout.fragment_about, container, false);

        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        InsetsManager.applyStatusBarInsets(binding.appbarLayout);
        InsetsManager.applyNavigationBarInsets(binding.content);

        // Setup back navigation
        NavController navController = Navigation.findNavController(view);
        AppBarConfiguration appBarConfiguration =
                ((MainActivity)requireActivity()).getAppBarConfiguration();
        NavigationUI.setupWithNavController(binding.toolbar, navController, appBarConfiguration);
    }
}

File: app/src/main/java/com/damhoe/scoresheetskat/app/about/AboutFragment.kt
package com.damhoe.scoresheetskat.app.about

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation.findNavController
import androidx.navigation.ui.NavigationUI.setupWithNavController
import com.damhoe.scoresheetskat.MainActivity
import com.damhoe.scoresheetskat.R
import com.damhoe.scoresheetskat.databinding.FragmentAboutBinding
import com.damhoe.scoresheetskat.shared_ui.utils.InsetsManager

class AboutFragment : Fragment() {

    lateinit var binding: FragmentAboutBinding

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentAboutBinding.inflate(layoutInflater)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        applyInsets()
        setUpNavigation()
    }

    private fun applyInsets() {
        binding.run {
            InsetsManager.applyStatusBarInsets(appbarLayout)
            InsetsManager.applyNavigationBarInsets(content)
        }
    }

    private fun setUpNavigation() {
        binding.run {
            val navController = findNavController()
            val appBarConfiguration = (requireActivity() as MainActivity).appBarConfiguration
            setupWithNavController(toolbar, navController, appBarConfiguration)
        }
    }

    private fun findNavController() =
        findNavController(requireActivity(), R.id.nav_host_fragment)
}